pkg/cloudprovider/fake/instancetype.go

```diff
@@ -80,22 +74,43 @@ func NewInstanceType(options InstanceTypeOptions) *InstanceType {
 	if options.OperatingSystems.Len() == 0 {
 		options.OperatingSystems = utilsets.NewString(string(v1.Linux), string(v1.Windows), "darwin")
 	}
-
-	return &InstanceType{
-		options: InstanceTypeOptions{
-			Name:             options.Name,
-			Offerings:        options.Offerings,
-			Architecture:     options.Architecture,
-			OperatingSystems: options.OperatingSystems,
-			Resources:        options.Resources,
-			Overhead:         options.Overhead,
+	i := &cloudprovider.InstanceType{
```

Comment:
optional: move requirements up and return this directly. Pure function way to represent this ;)